File: motomanager/frontend/motomanager-web/src/app/models/vehicle.ts
export interface Vehicle {
  id: number;
  registration: string;
  make: string;
  model: string;
  year?: number;
  isActive: boolean;
  clientId?: number;
  clientName?: string;
  createdAt: string;
  updatedAt: string;
}

export interface CreateVehicleRequest {
  registration: string;
  make: string;
  model: string;
  year?: number;
  clientId?: number;
}

export interface UpdateVehicleRequest {
  registration: string;
  make: string;
  model: string;
  year?: number;
  isActive: boolean;
  clientId?: number;
}

File: motomanager/backend/MotoManager.Domain/Entities/Vehicle.cs
namespace MotoManager.Domain.Entities;

public class Vehicle
{
    public long Id { get; set; }
    public string Registration { get; set; } = string.Empty;
    public string Make { get; set; } = string.Empty;
    public string Model { get; set; } = string.Empty;
    public int? Year { get; set; }
    public bool IsActive { get; set; } = true;
    public DateTimeOffset CreatedAt { get; set; }
    public DateTimeOffset UpdatedAt { get; set; }

    public ICollection<ServiceOrder> ServiceOrders { get; set; } = new List<ServiceOrder>();
}
